### ISSUE TITLE
Error: Argument 1 ($courseid) must be of type int, string given

### ISSUE THREAD

**Issue by nolowolf** [edu-sharing/moodle-mod_edusharing#11]
When importing coursecontent from one course to the other, the following error occurs:

Exception - rebuild_course_cache(): Argument #1 ($courseid) must be of type int, string given, called in [dirroot]/mod/edusharing/classes/RestoreHelper.php on line 101

Reproduce the bug:

- create an empty course "testcourse 1"
- create another empty course "testcourse 2"
- got to testcourse 2 and via "course reuse" import "testcourse 1"
- click "jump to final step"

the following message is shown:

```
Exception - rebuild_course_cache(): Argument #1 ($courseid) must be of type int, string given, called in [dirroot]/mod/edusharing/classes/RestoreHelper.php on line 101

[More information about this error](https://docs.moodle.org/403/en/error/moodle/generalexceptionmessage)

Debug info:
Error code: generalexceptionmessage×Dismiss this notification
Stack trace:
line 2822 of /lib/modinfolib.php: TypeError thrown
line 101 of /mod/edusharing/classes/RestoreHelper.php: call to rebuild_course_cache()
line 201 of /mod/edusharing/classes/observer.php: call to mod_edusharing\RestoreHelper->convert_inline_options()
line ? of unknownfile: call to mod_edusharing_observer::course_restored()
line 155 of /lib/classes/event/manager.php: call to call_user_func()
line 75 of /lib/classes/event/manager.php: call to core\event\manager::process_buffers()
line 795 of /lib/classes/event/base.php: call to core\event\manager::dispatch()
line 195 of /backup/util/plan/restore_plan.class.php: call to core\event\base->trigger()
line 411 of /backup/controller/restore_controller.class.php: call to restore_plan->execute()
line 193 of /backup/import.php: call to restore_controller->execute_plan()
```

**Comment by klopfa10** [comment]
Same problem here. 
It seems like the import is working - error message appears at 100%. 

During another import the same message came up at 75%, but instead of courseID it was courseSummary (string expected - null given). This import did not fully work.

**Comment by zieglermarian** [comment]
Thanks for finding the issues. They are fixed in the newest version of the plugin (8.0.9) which is available in the [Moodle Plugins Directory](https://moodle.org/plugins/mod_edusharing)
